[Issue_72] Remove unneeded passing of ATTRIBUTES to the AbstractBoott…

bstansberry opened the pull request:
…imeAddStepHandler c'tor

Resolves #72

WolfgangHG commented:
Is it really this trivial ;-)? Would attributes defined in https://github.com/wildfly/wildfly-archetypes/blob/d91f67d3dc80accf2d2b28d671bc5086a8f2cf6a/wildfly-subsystem-archetype/src/main/resources/archetype-resources/src/main/java/SubsystemDefinition.java#L15 still work? 
The "JaxrsSubsystemDefinition" that I linked in #72 added an override "registerAttributes", but it also has a different base class.

I don't know enough of the subsystem internals, so I don't have a sample for attributes.

bstansberry commented:
@WolfgangHG Yes, it's really that trivial. :-)

The 'registerAttributes' stuff you see in JaxrsSubsystemDefinition is handled for the archetype-generated subsystem by SubsystemDefinition's superclass.

Because the attributes are registered with the ManagementResourceRegistration, a typical add handler really shouldn't need to have the attributes provided to it via it's constructor, as it can get them from the ManagementResourceRegistration. So we implemented that and we've been deprecating and later removing code that assumes it's necessary to pass in attributes.

I created a project from the archetype and added an attribute to SubsystemDefinition.ATTRIBUTES (and other necessary places unrelated to this issue) and it works fine.

bstansberry commented:
Thanks @WolfgangHG and @jamezp
